### ISSUE TITLE
Sometimes highlights aren't cleared

### ISSUE THREAD

**Issue by braver** [SublimeLinter/SublimeLinter#883]
<img width="264" alt="screen shot 2018-01-24 at 14 24 34" src="https://user-images.githubusercontent.com/2543659/35334364-695e2a8e-0112-11e8-80d9-48a6be3a5040.png">

We used to run into this early on, but haven't seen them much since. I don't know what happens or how to recreate it. The status bar and the panel correctly display 0 problems.

**Comment by pykong** [comment]
This means the highlight zombies are back then? I have not yet familiarized with the changes @kaste  introduced to `highlight.py`. In one of our discussions I raised the possibility that removing the persistence of highlights across SubliemeLinter sessions, may eventually bring zombies back. That said it is an unlikely cause for the return, but a possibility we should still investigate.

See: https://github.com/SublimeLinter/SublimeLinter/pull/766/files/4e7cfcdf9798620f44c5d739e6eedb311787ffda#r159756877

**Comment by braver** [comment]
Not sure yet, perhaps the state of my install of SL wasn't clean at that point. I seem to find this when using tslint mostly, will look out for 👻 the next few days.

**Comment by kaste** [comment]
I've seen them today for the first time. So this could mean I introduced them yesterday.

But, I did a lot of git checkouts, rebases, so SublimeLinter will naturally and expected get in an unknown state.

**Comment by kaste** [comment]
I did remove one `clear` call, when you change settings. These are not zombie highlights, they get removed if you just type away, on the next lint.

**Comment by pykong** [comment]
This confirms the minor nature of this bug. Yet, I prefer those highlights getting cleared on settings change nonetheless.

**Comment by kaste** [comment]
@pykong Do you really want that highlights get cleared when settings change? If yes, please explain.

**Comment by braver** [comment]
Most settings probably affect the way highlights should be drawn in some way, so if a setting changes it's probably not a bad idea to reset completely.

Must say I haven't had this problem since.

**Comment by pykong** [comment]
@kaste 
> Most settings probably affect the way highlights should be drawn in some way, so if a setting changes it's probably not a bad idea to reset completely.

This.

**Comment by kaste** [comment]
Seriously, that's sloppy language. Write it down, like a spec or user story or whatever, and we implement it.

This issue is about something very serious, we can't replicate. And I only saw it *once*.

The commit I linked here is something I immediately remembered from the last day. From your speaking, I don't even know if you can describe the previous and the current behavior after this change. I actually tested this change btw.

**Comment by braver** [comment]
Well whatever about that change, changing settings has nothing to do with any of the cases where I encountered the issue at hand. When we seemed to have a need for nuking the highlights after changing settings, SL was in a completely different state than now. Most of it was unknown and a lot of it was a mess. I tested changing settings and it’s well behaved these days.

Also haven’t seen it anymore, so let’s not waste energy on it till we have a good steps-to-reproduce.
